lldb/source/Plugins/Disassembler/LLVMC/DisassemblerLLVMC.cpp

```diff
@@ -1446,6 +1446,33 @@ bool DisassemblerLLVMC::MCDisasmInstance::IsAuthenticated(
   return InstrDesc.isAuthenticated() || IsBrkC47x;
 }
 
+void DisassemblerLLVMC::UpdateFeatureString(llvm::StringRef additional_features,
+                                            std::string &features) {
+  // Allow users to override default additional features.
+  for (llvm::StringRef flag : llvm::split(additional_features, ",")) {
+    flag = flag.trim();
+    if (flag.empty())
+      continue;
+    // Check if disable (-c) is already present before adding default enable
+    // (+c), since enable (+c) takes precedence whether disable (-c) appears
```

Comment:
I don't understand the logic of this comment or block.  This code comes down to

```
    if (flag.starts_with('+')) {
      if (features.find(disable_flag) == std::string::npos) {
        // -flag is not present
        if (!features.empty())
          features = ',' + features;
        }
        features = flag.str() + features;
      } 
         /// What about when -flag IS present?  We ignore +flag, is that the idea?
    } else {
      if (!features.empty())
        features = ',' + features;
      }
      features = flag.str() + features;
```

What if we have a +flag in features already, and additional_features has `-flag`?  Do we need to remove `+flag` from features?  Given that we drop +flag if -flag is present, it seems like we must do that?

Reply:
I still wonder about using a std::string representation in ArchSpec is the right answer, if we need to manage things like this properly.  It seems like a more natural data structure would be a set<string> and remove/add entries following the rules you've intended.  This also avoids issues of features being added multiple times, which I'm not sure this patch avoids yet, but I should re-read that more clearly.  I'm not saying you should rewrite in terms of a set and have a "GetAsString" method to return a std::string of the features, but either it's important to do this correctly or not -- and using a comma separated string is making it all a lot easier to do incorrectly than it should be.